### ISSUE TITLE
Network manager support: ConnMan

### ISSUE THREAD

**Issue by koraa** [rosenpass/rosenpass#82]
https://wiki.archlinux.org/title/ConnMan

**Comment by algora-pbc** [comment]
## 💎 $150 bounty [• Rosenpass](https://algora.io/rosenpass)
### Steps to solve:
1. **Start working**: Comment `/attempt #82` with your implementation plan
2. **Submit work**: Create a pull request including `/claim #82` in the PR body to claim the bounty
3. **Receive payment**: 100% of the bounty is received 2-5 days post-reward. [Make sure you are eligible for payouts](https://algora.io/docs/payments#supported-countries-regions)

### ❗ Important guidelines:
- To claim a bounty, you need to **provide a short demo video** of your changes in your pull request
- If anything is unclear, **ask for clarification** before starting as this will help avoid potential rework
- Low quality AI PRs will not receive review and will be closed
- Do not ask to be assigned unless you've contributed before

Thank you for contributing to rosenpass/rosenpass!

| Attempt | Started (UTC) | Solution | Actions |
| --- | --- | --- | --- |
| 🟢 @a4to | Mar 02, 2026, 02:42:23 AM | #714 | [Reward](https://algora.io/claims/GvfyQk2CFQUjCFZa) |
| 🟢 @algora-pbc | Mar 02, 2026, 02:42:24 AM | WIP |  |
| 🟢 @pqcfox | Jan 07, 2024, 09:28:52 PM | WIP |  |
| 🔴 @aazam-gh | Sep 09, 2023, 09:17:50 AM | WIP |  |
| 🟢 @davidequarracino | Mar 10, 2026, 02:10:26 AM | #718 | [Reward](https://algora.io/claims/3RERdmUoNAHSycFy) |
| 🟢 @blessuselessk | Mar 21, 2026, 04:52:40 AM | #722 | [Reward](https://algora.io/claims/4KMkqQkNpWn1d7rd) |
| 🟢 @zzinterest | Mar 29, 2026, 12:32:23 PM | #724 | [Reward](https://algora.io/claims/HgYi1G6235x5VFiU) |

**Comment by aazam-gh** [comment]
is rosenpass available for ubuntu?

**Comment by emilengler** [comment]
> is rosenpass available for ubuntu? 

You have to compile it for yourself.

**Comment by aazam-gh** [comment]
Would I need an arch Linux distro to solve this issue?

**Comment by emilengler** [comment]
> Would I need an arch Linux distro to solve this issue?

No. You can clone the repository and compile the binary using `cargo build --release`.

**Comment by pqcfox** [comment]
I'd love to make an attempt at this!

/attempt #82

**Comment by aazam-gh** [comment]
> No. You can clone the repository and compile the binary using `cargo build --release`.
I did that but I am not finding any binary file under the target directory for rp

**Comment by BenraouaneSoufiane** [comment]
Also you want two shell scripts to init connection & auto support connman or you want to edit the core implimentation (rust files)?

**Comment by koraa** [comment]
@BenraouaneSoufiane I don't think there should be a basic issue with running ConnMan + Rosenpass (not managed by ConnMan), but I may be wrong.

This issue is about integrating Rosenpass-support into ConnMan, through a plugin or through whatever means so ConnMan users can configure Rosenpass-based WireGuard connections through ConnMan.

**Comment by BenraouaneSoufiane** [comment]
@koraa I've created the plugin, but still stucked with some issue (connmanctl connect vpn_somehost creates the tunnel, rp shared keys will shows in both client/server, but the command does not return connected) if you'll review it (means you're free not busy) I'll continue fix it

**Comment by ilka-schulz** [comment]
We are closing this issue for now and might attempt it later. We received several low-quality LLM PRs for our algora-bounty issues and need to work through a huge amount of slop first.
